stream.go

```diff
@@ -642,10 +663,27 @@ func writeCell(buf *bufferedWriter, c xlsxC) {
 func (sw *StreamWriter) writeSheetData() {
 	if !sw.sheetWritten {
 		bulkAppendFields(&sw.rawData, sw.worksheet, 4, 5)
-		if sw.cols.Len() > 0 {
-			_, _ = sw.rawData.WriteString("<cols>")
-			_, _ = sw.rawData.WriteString(sw.cols.String())
-			_, _ = sw.rawData.WriteString("</cols>")
+		if sw.worksheet.Cols != nil {
+			for _, col := range sw.worksheet.Cols.Col {
+				_, _ = sw.rawData.WriteString("<cols>")
```

Comment:
Note that these changes will cause stream writers to write redundant cols element, fixed by pull request #2079.